src/components/LHNOptionsList/OptionRowLHN.js

```diff
@@ -0,0 +1,220 @@
+                                    tooltipEnabled
+                                    numberOfLines={1}
+                                    textStyles={displayNameStyle}
+                                    shouldUseFullTitle={optionItem.isChatRoom || optionItem.isPolicyExpenseChat}
+                                />
+                                {optionItem.alternateText ? (
+                                    <Text
+                                        style={alternateTextStyle}
+                                        numberOfLines={1}
+                                    >
+                                        {optionItem.alternateText}
+                                    </Text>
+                                ) : null}
+                            </View>
+                            {optionItem.descriptiveText ? (
+                                <View style={[styles.flexWrap]}>
+                                    <Text style={[styles.textLabel]}>
+                                        {optionItem.descriptiveText}
+                                    </Text>
+                                </View>
+                            ) : null}
+                            {optionItem.brickRoadIndicator === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR && (
+                                <View style={[styles.alignItemsCenter, styles.justifyContentCenter]}>
+                                    <Icon
+                                        src={Expensicons.DotIndicator}
+                                        fill={colors.red}
+                                        height={variables.iconSizeSmall}
+                                        width={variables.iconSizeSmall}
+                                    />
+                                </View>
+                            )}
+                        </View>
+                    </View>
+                    <View style={[styles.flexRow, styles.alignItemsCenter]}>
+                        {optionItem.hasDraftComment && (
+                            <View style={styles.ml2}>
+                                <Icon src={Expensicons.Pencil} height={16} width={16} />
+                            </View>
+                        )}
+                        {optionItem.hasOutstandingIOU && (
+                            <IOUBadge iouReportID={optionItem.iouReportID} />
+                        )}
+                        {optionItem.isPinned && (
+                            <View style={styles.ml2}>
+                                <Icon src={Expensicons.Pin} height={16} width={16} />
+                            </View>
+                        )}
+                    </View>
+                </TouchableOpacity>
+            )}
+        </Hoverable>
+    );
+};
+
+OptionRowLHN.propTypes = propTypes;
+OptionRowLHN.defaultProps = defaultProps;
+OptionRowLHN.displayName = 'OptionRowLHN';
+
+export default withLocalize(OptionRowLHN);
```

Comment:
Is React.memo is missing is it on purpose?

Reply:
Sort of, yes. I didn't purposefully avoid it, but could help me understand how it helps here?

Reply:
FlatList rerenders items onScroll event also when you add more items. Without memo items will be re-rendered.

Reply:
My concern is the option row depends on 6 different collections of data from Onyx in order to render, and it should re-render if any of those change (like the name of a report participant for example). Since the only prop being passed to this component is `reportID`, then the component has no other way of re-rendering when some of the Onyx data changes unless `OptionRowLHN` also subscribes to the same `withOnyx()` data. I was really trying to avoid having possibly thousands of these `OptionRowLHN` components all connected to Onyx individually as I don't know what it would do for performance.

Ultimately, if it's very very cheap to render `OptionRowLHN` then the benefit of memoizing it becomes less.

Can you think of anything else that would help?